iroh/src/magicsock/endpoint_map/endpoint_state.rs

```diff
@@ -130,7 +131,7 @@ pub(super) struct EndpointStateActor {
     /// holepunching regularly.
     ///
     /// We only select a path once the path is functional in Quinn.
-    selected_path: Option<transports::Addr>,
+    selected_path: Watchable<Option<TransportAddr>>,
```

Comment:
Oh right, `transports::Addr` vs `TransportAddr` is the reason why you have two fields before?  Yeah, not sure if I want the code here to have to deal with this all the time.  The code in this actor has to be as simple and logical as possible.

Is the conversion from `transports::Addr` -> `TransportAddr` infallible?  Can the user-facing watcher not use a `Watcher::map` on this?  (I've not tried to use that feature myself before so could be off the mark)